rust/agama-dbus-server/src/questions/answers.rs

```diff
@@ -0,0 +1,291 @@
+use std::collections::HashMap;
+
+use anyhow::Context;
+use serde::{Deserialize, Serialize};
+
+/// Data structure for single yaml answer. For variables specification see
+/// corresponding [agama_lib::questions::GenericQuestion] fields.
+/// The *matcher* part is: `class`, `text`, `data`.
+/// The *answer* part is: `answer`, `password`.
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+struct Answer {
+    pub class: Option<String>,
+    pub text: Option<String>,
+    /// A matching GenericQuestion can have other data fields too
+    pub data: Option<HashMap<String, String>>,
+    /// The answer text is the only mandatory part of an Answer
+    pub answer: String,
+    /// All possible mixins have to be here, so they can be specified in an Answer
```

Comment:
Mixins?

Reply:
Mixins to Base Question like `WithPassword`. So it means that all possible additional data entries from additional mixins to question have to be specified in answer.